builtin/logical/transit/path_cache_config.go

```diff
@@ -0,0 +1,124 @@
+package transit
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/helper/keysutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func (b *backend) pathCacheConfig() *framework.Path {
+	return &framework.Path{
+		Pattern: "cache-config",
+		Fields: map[string]*framework.FieldSchema{
+			"cache-type": &framework.FieldSchema{
+				Type:     framework.TypeString,
+				Required: true,
+				Description: `
+Type of cache to use. Currently "syncmap" and "lru" are supported.
```

Comment:
I'm not sure about the term "syncmap", as it seems quite coupled to the Go stdlib and isn't a generally used term. Plus we might switch the implementation but retain the behavior. From a customer perspective, I know what "lru" implies, but what are the characteristics of what is now "syncmap"? Maybe simplifying to "map" is enough, but I'd be curious what others think.

Reply:
Good points.  How about "unlimited"?

Reply:
I think a better UX would be removing the `cache-type` parameter all together and just require a `cache-size` to be set. It doesn't really matter to the user what data structure is used under the hood. The default can be `0` which means unlimited (syncmap) and if `>0` we use the LRU

Reply:
@briankassouf I was just thinking something similar here: https://github.com/hashicorp/vault/pull/6225#discussion_r261336646